src/vmspawn/vmspawn.c

```diff
@@ -1248,10 +1293,37 @@ static int run_virtual_machine(int kvm_device_fd, int vhost_device_fd) {
                         return log_oom();
         }
 
-        if (arg_initrd) {
-                r = strv_extend_many(&cmdline, "-initrd", arg_initrd);
+        char *initrd = NULL;
+        _cleanup_(rm_rf_physical_and_freep) char *merged_initrd = NULL;
+        size_t n_initrds = strv_length(arg_initrds);
+        if (n_initrds == 1)
+                initrd = arg_initrds[0];
+        else if (n_initrds > 1) {
+                r = merge_initrds(&merged_initrd);
+                if (r < 0)
+                        return r;
+
+                initrd = merged_initrd;
+        }
+
+        if (initrd) {
+                r = strv_extend_many(&cmdline, "-initrd", initrd);
                 if (r < 0)
                         return log_oom();
+        } else if (kernel) {
+                KernelImageType kt;
+                r = inspect_kernel(
+                                AT_FDCWD,
+                                kernel,
+                                &kt,
+                                /* ret_cmdline= */ NULL,
+                                /* ret_uname= */ NULL,
+                                /* ret_pretty_name= */ NULL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to inspect the type of the kernel: %m");
```

Comment:
This message is useless without the filename.

Reply:
This code should actually be removed https://github.com/systemd/systemd/pull/29960#discussion_r1484246048
I seem to have added it back by accident in my merging initrds commit.